docs/docs/crafty.md

```diff
@@ -3,4 +3,184 @@ id: crafty
+
+## Project setup
+
+This guide will show you the process of making the contracts in the Crafty game upgradeable by using ZeppelinOS, which we'll install globally to have it available on all our projects.
+
+```sh
+npm install --global zos
+```
+
+All snippets here were extracted from the public [Crafty repository](https://github.com/zeppelinos/crafty).
+
+## Contracts
+
+Crafty works with only two contracts:
+* [`Crafty`](https://github.com/zeppelinos/crafty/blob/master/contracts/Crafty.sol) itself, containing the game logic (how new recipes are added, how crafting takes place, etc.) plus some role-based access control ([RBAC](https://github.com/OpenZeppelin/openzeppelin-solidity/blob/zos-release/contracts/ownership/rbac/RBAC.sol), powered by [OpenZeppelin](https://openzeppelin.org/)), for admin tasks, such as deletion of tokens.
+* The [`CraftableToken`](https://github.com/zeppelinos/crafty/blob/master/contracts/CraftableToken.sol)s, which are a simple extension of a mintable ERC20, holding token creator data, a URI for metadata, and a list of required ingredients, along with the amounts for each.
+
+To better see how these two contracts interact, take a look at `Crafty`'s `craft` method, where a user provides the address of a `CraftableToken` to craft, and `Crafty` makes sure the required ingredients are consumed before minting it.
+
+```
+function craft(CraftableToken _craftable) public {
+  address player = msg.sender;
+
+  uint256 totalSteps = _craftable.getTotalRecipeSteps();
+  for (uint i = 0; i < totalSteps; ++i) {
+    ERC20 ingredient;
+    uint256 amountNeeded;
+    (ingredient, amountNeeded) = _craftable.getRecipeStep(i);
+
+    ingredient.transferFrom(player, address(this), amountNeeded);
+  }
+
+  // Issue the crafted token - for ingredient-less tokens, a higher number of
+  // tokens are minted.
+  uint256 tokensToMint = (totalSteps == 0) ? 100 : 1;
+  _craftable.mint(player, tokensToMint);
+}
+```
+
+## Developing with ZeppelinOS
+
+So, how would we go about integrating ZeppelinOS to such a project? The process is remarkably simple: we'll cover it step by step. But before we get started, don't forget to initialize your project with ZeppelinOS:
+
+```sh
+zos init Crafty
+```
+
+This will create a `zos.json` file, which will store how your project is structured in ZeppelinOS: you'll probably want to commit this file.
+
+### Making `Crafty` upgradeable
+
+There are multiple reasons why we would want to be able to upgrade `Crafty`'s game logic contract: from bug-fixing, to adding new functionalities. Remember how `Crafty` uses RBAC (role-based access control)? An upgrade could easily add new roles, such as a `curator` role, which would be in charge of approving new tokens before they are added to the game, or highlighting featured creations to be displayed on the front page. These simple but useful extensions would not be possible without ZeppelinOS.
```

Comment:
Isn't adding a role a dynamic operation which RBAC.sol can handle without the need to alter the code? If no, then maybe we could be more explicit on the type of update that would require this.

Reply:
[`addRole` is an internal function](https://github.com/OpenZeppelin/openzeppelin-solidity/blob/zos-release/contracts/ownership/rbac/RBAC.sol#L57), so you'd need to add external functions that call those, with modifiers to restrict access (like [here](https://github.com/zeppelinos/crafty/blob/master/contracts/Crafty.sol#L35)).